### PR TITLE
Consent flow

### PR DESCRIPTION
**Pull request by TsankoBg**
- Add consent pop flow
- Fix sendSyncUrl
- Fix product review sync

[sc-30859]

### CODE REVIEW

File: common/src/Module.php [hunk 1]
```diff
@@ -175,7 +161,27 @@ public function hookHeader($params) {
         ]);
     }
 
-    public function hookFooter($params) {
+
+    public function hookDisplayOrderConfirmation($params) {
+        $order = $params['order'];
+        $customer = new \Customer((int) ($order->id_customer));
```
Comment (apeschar) [inline common/src/Module.php]:
```suggestion
        $customer = new \Customer((int) $order->id_customer);
```

~~~
> $x = (object) ['y' => '1'];
= {#5209
    +"y": "1",
  }

> (int) $x->y;
= 1

> (int) ($x->y);
= 1
~~~

File: common/src/Module.php [hunk 2]
```diff
@@ -354,22 +360,26 @@ private function sendInvites($ps_shop_id) {
         }
 
         foreach ($orders as $order) {
+            if (
+                Configuration::get($this->getConfigName('INVITE'), null, null, $ps_shop_id) == 3
+                && !$this->hasConsent($order['id_order'], $ps_shop_id)
+            ) {
+                $this->markInviteAsSent($order['id_order']);
+                PrestaShopLogger::addLog(
+                    sprintf('Invitation was not created for order (%s) as customer did not give a consent', $order['id_order']),
```
Comment (apeschar) [inline common/src/Module.php]:
```suggestion
                    sprintf('Invitation was not created for order (%s) as customer did not consent', $order['id_order']),
```

File: common/src/Module.php [hunk 3]
```diff
@@ -614,28 +624,29 @@ private function getPluginTableName($name) {
...
             CURLOPT_RETURNTRANSFER => true,
             CURLOPT_FOLLOWLOCATION => true,
+            CURLOPT_CUSTOMREQUEST => $method,
         ];
-        if (!curl_setopt_array($ch, $options)) {
-            throw new RuntimeException('curl_setopt_array failed');
+        $ch = $this->getCurl($default_options + $options);
+        if (!$ch) {
+            throw new RuntimeException('curl_init failed');
         }
```
Comment (apeschar) [inline common/src/Module.php]: This should go in `getCurl` or be dropped altogether.

File: common/src/Module.php [hunk 4]
```diff
@@ -614,28 +624,29 @@ private function getPluginTableName($name) {
 
     /**
      * @param string $url
+     * @param string $method
      * @param array $options
      * @return string
      */
```
Comment (apeschar) [inline common/src/Module.php]:
```suggestion
```

File: common/src/Module.php [hunk 5]
```diff
@@ -614,28 +624,29 @@ private function getPluginTableName($name) {
...
-    private function request($url, $options = []) {
-        $ch = curl_init($url);
-        if (!$ch) {
-            throw new RuntimeException('curl_init failed');
-        }
-        $options += [
+    private function request(string $url, string $method, array $options = []) {
```
Comment (apeschar) [inline common/src/Module.php]:
```suggestion
    private function request(string $url, string $method, array $options = []): string {
```

File: common/src/Sync.php [hunk 6]
```diff
@@ -53,8 +53,16 @@ private function syncProductReview(array $product_review): void {
         $entity_manager = $this->container->get('doctrine.orm.entity_manager');
         $date_add = DateTime::createFromFormat('Y-m-d H:i:s', $product_review['created']);
 
-        $product_comment_repository = $entity_manager->getRepository(ProductComment::class);
-        $product_comment = $product_review['id'] ? $product_comment_repository->find($product_review['id']) : new ProductComment();
+        if ($product_review['id']) {
+            $product_comment_repository = $entity_manager->getRepository(ProductComment::class);
+            $product_comment = $product_comment_repository->find($product_review['id']);
+            if (!$product_comment) {
+                $this->returnResponseCode(404, sprintf('Could not find product review with ID (%d)', $product_review['id']));
```
Comment (apeschar) [inline common/src/Sync.php]:
```suggestion
                $this->returnResponseCode(404, sprintf('Could not find product review with ID %d', $product_review['id']));
```

**Review by apeschar — CHANGES_REQUESTED** [review]

File: common/src/Module.php (hunk 1 above)
Comment (TsankoBg) [reply common/src/Module.php]: ah I am blind ... I did not notice the `()`

**Review by apeschar — APPROVED** [review]

**Pull request merged**